### ISSUE TITLE
android.os.FileUriExposedException with file:// sound (Android 7)

### ISSUE THREAD

**Issue by matijse** [katzer/cordova-plugin-local-notifications#1181]
When scheduling a notification on Android 7, using a `file://` url for the sound, I get an `android.os.FileUriExposedException` when the notification takes place. 

This is only on Android 7 and up, as confirmed by https://developer.android.com/reference/android/os/FileUriExposedException.html

Code

```
cordova.plugins.notification.local.schedule({
      id: 1,
      title: "title",
      text: "text",
      sound: "file://assets/sounds/dream.mp3",
      at: date
});
```

Complete error message

```
01-04 12:39:32.178: E/AndroidRuntime(15198): java.lang.RuntimeException: Unable to start receiver de.appplant.cordova.plugin.localnotification.TriggerReceiver: android.os.FileUriExposedException: file:///storage/emulated/0/Android/data/com.app.id/cache/localnotification/dream.mp3 exposed beyond app through Notification.sound
01-04 12:39:32.178: E/AndroidRuntime(15198): 	at android.app.ActivityThread.handleReceiver(ActivityThread.java:3018)
01-04 12:39:32.178: E/AndroidRuntime(15198): 	at android.app.ActivityThread.-wrap18(ActivityThread.java)
01-04 12:39:32.178: E/AndroidRuntime(15198): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1544)
01-04 12:39:32.178: E/AndroidRuntime(15198): 	at android.os.Handler.dispatchMessage(Handler.java:102)
01-04 12:39:32.178: E/AndroidRuntime(15198): 	at android.os.Looper.loop(Looper.java:154)
01-04 12:39:32.178: E/AndroidRuntime(15198): 	at android.app.ActivityThread.main(ActivityThread.java:6077)
01-04 12:39:32.178: E/AndroidRuntime(15198): 	at java.lang.reflect.Method.invoke(Native Method)
01-04 12:39:32.178: E/AndroidRuntime(15198): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:865)
01-04 12:39:32.178: E/AndroidRuntime(15198): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:755)
01-04 12:39:32.178: E/AndroidRuntime(15198): Caused by: android.os.FileUriExposedException: file:///storage/emulated/0/Android/data/com.app.id/cache/localnotification/dream.mp3 exposed beyond app through Notification.sound
01-04 12:39:32.178: E/AndroidRuntime(15198): 	at android.os.StrictMode.onFileUriExposed(StrictMode.java:1799)
01-04 12:39:32.178: E/AndroidRuntime(15198): 	at android.net.Uri.checkFileUriExposed(Uri.java:2346)
01-04 12:39:32.178: E/AndroidRuntime(15198): 	at android.app.NotificationManager.notifyAsUser(NotificationManager.java:300)
01-04 12:39:32.178: E/AndroidRuntime(15198): 	at android.app.NotificationManager.notify(NotificationManager.java:284)
01-04 12:39:32.178: E/AndroidRuntime(15198): 	at android.app.NotificationManager.notify(NotificationManager.java:268)
01-04 12:39:32.178: E/AndroidRuntime(15198): 	at de.appplant.cordova.plugin.notification.Notification.showNotification(Notification.java:239)
01-04 12:39:32.178: E/AndroidRuntime(15198): 	at de.appplant.cordova.plugin.notification.Notification.show(Notification.java:224)
01-04 12:39:32.178: E/AndroidRuntime(15198): 	at de.appplant.cordova.plugin.notification.TriggerReceiver.onTrigger(TriggerReceiver.java:45)
01-04 12:39:32.178: E/AndroidRuntime(15198): 	at de.appplant.cordova.plugin.localnotification.TriggerReceiver.onTrigger(TriggerReceiver.java:48)
01-04 12:39:32.178: E/AndroidRuntime(15198): 	at de.appplant.cordova.plugin.notification.AbstractTriggerReceiver.onReceive(AbstractTriggerReceiver.java:75)
01-04 12:39:32.178: E/AndroidRuntime(15198): 	at android.app.ActivityThread.handleReceiver(ActivityThread.java:3011)
01-04 12:39:32.178: E/AndroidRuntime(15198): 	... 8 more
```

**Comment by mherb63** [comment]
same here...any work-arounds?

**Comment by Sohib** [comment]
same here

**Comment by chevarria** [comment]
same here

**Comment by souly1** [comment]
+1

**Comment by davidlr99** [comment]
+1

**Comment by vegham** [comment]
See my comment for work-around: https://github.com/katzer/cordova-plugin-local-notifications/issues/1261#issuecomment-298097708

**Comment by matijse** [comment]
Some time ago I started trying to fix this by adopting the content:// protocol as @vegham said in his comment. You can find my progress at https://github.com/matijse/cordova-plugin-local-notifications/commit/b75cbf46affa7096929a94a4209cb22e7bd7e0b0, but that didn't work yet. I haven't had time to finish it, but it might be a starting point for someone else...

**Comment by katzer** [comment]
Has been fixed with 0.9-beta
